### ISSUE TITLE
h5py broken in recent pytorch images

### ISSUE THREAD

**Issue by sparticlesteve** [NERSC/nersc-ml-images#11]
@ShashankSubramanian reported an issue with recent nersc pytorch images:

```
> shifter --image=nersc/pytorch:ngc-24.05-v1 python -c 'import h5py'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/h5py/__init__.py", line 45, in <module>
    from ._conv import register_converters as _register_converters, \
  File "h5py/_conv.pyx", line 1, in init h5py._conv
  File "h5py/h5r.pyx", line 1, in init h5py.h5r
  File "h5py/h5p.pyx", line 1, in init h5py.h5p
  File "h5py/h5t.pyx", line 235, in init h5py.h5t
  File "h5py/h5t.pyx", line 80, in h5py.h5t.lockid
  File "h5py/h5t.pyx", line 49, in h5py.h5t.typewrap
ValueError: Not a datatype (not a datatype)
```
I confirmed it also with 24.06.00.
I also confirmed that the podman-hpc image `nersc/pytorch:ngc-24.05-v1` is affected.
Sounds related: https://github.com/h5py/h5py/issues/2436

**Comment by sparticlesteve** [comment]
I'm testing reverting HDF5 to an older version.
We've been using the latest one, `1.14.4.3`, whereas had no troubles with images that used `1.12.3`.

**Comment by sparticlesteve** [comment]
Resolved with 0376a51fac5a439426f7e008c282983f3091387a
